airbyte-integrations/connectors/source-mysql/src/main/java/io/airbyte/integrations/source/mysql/MySqlQueryUtils.java

```diff
@@ -0,0 +1,73 @@
+package io.airbyte.integrations.source.mysql;
+
+import static io.airbyte.integrations.source.relationaldb.RelationalDbQueryUtils.getFullyQualifiedTableNameWithQuoting;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.google.common.base.Preconditions;
+import io.airbyte.db.jdbc.JdbcDatabase;
+import io.airbyte.db.jdbc.JdbcUtils;
+import io.airbyte.protocol.models.AirbyteStreamNameNamespacePair;
+import io.airbyte.protocol.models.v0.ConfiguredAirbyteStream;
+import java.sql.SQLException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class MySqlQueryUtils {
+  private static final Logger LOGGER = LoggerFactory.getLogger(MySqlQueryUtils.class);
+  public record TableSizeInfo(Long tableSize, Long avgRowLength) { }
+
+  public static final String TABLE_ESTIMATE_QUERY =
+      """
+      SELECT
+        (data_length + index_length) as %s,
+        AVG_ROW_LENGTH as %s
+     FROM
+        information_schema.tables
+     WHERE
+        table_schema = '%s' AND table_name = '%s'; 
+      """;
+
+  public static final String TABLE_SIZE_BYTES_COL = "TotalSizeBytes";
+  public static final String AVG_ROW_LENGTH = "AVG_ROW_LENGTH";
+
+  public static Map<AirbyteStreamNameNamespacePair, TableSizeInfo> getTableSizeInfoForStreams(final JdbcDatabase database,
```

Comment:
Can you add a test for this method?

Reply:
Ah I tried, but the issue here is that this query is just an estimate. This is to get a good idea without incurring the cost of the accurate `select count(*)` query. It yields (slightly) different results, so skipping this since it isn't 100% deterministically

Reply:
Ah okay. The e2e test that's already there should be good then